browser/src/control/Control.JSDialog.js

```diff
@@ -383,15 +383,22 @@ L.Control.JSDialog = L.Control.extend({
 		if (instance.canHaveFocus && initialFocusElement && initialFocusElement.length)
 			initialFocusElement[0].focus();
 
-		var focusWidget = instance.init_focus_id ? instance.container.querySelector('[id=\'' + instance.init_focus_id + '\']') : null;
-		if (focusWidget)
-			focusWidget.focus();
-		if (focusWidget && document.activeElement !== focusWidget) {
-			var firstFocusable = JSDialog.GetFocusableElements(focusWidget);
-			if (firstFocusable && firstFocusable.length)
-				firstFocusable[0].focus();
-			else
-				console.error('cannot get focus for widget: "' + instance.init_focus_id + '"');
+		// Check if the element with id 'tabcontrol' exists in the DOM
+		const isTabControlWidget = document.getElementById('tabcontrol') ? true : false;
+		if (isTabControlWidget) {
+			// Find the first focusable tab by its id 'tabcontrol-1'
+			const firstFocusableTab = document.getElementById('tabcontrol-1');
+			firstFocusableTab.focus();
+		}
+		else {
+			const focusWidget = instance.init_focus_id ? instance.container.querySelector('[id=\'' + instance.init_focus_id + '\']') : null;
```

Comment:
Here we have explicitly defined element which should be focused as wanted by the dialog designer.
I think it should be always prefered option.

What is the example of dialog/widget it didn't work?

Reply:
Menu "References" ( Dialog "Table of Contents")